### PR TITLE
fix: add Page.loadEventFired listener before navigating

### PR DESCRIPTION
**Pull request by Kriegslustig**
When navigating it sometimes happens that the navigate callback result is sent after the `Page.loadEventFired` event is sent. The result is that the PDF generation promise never resolves. With this PR the `generate` function first sets the `Page.loadEventFired` listener and calls `navigate` after.

For my personally at least 50% of the tests were failing locally every time. Which tests failed seemed pretty random. Now they consistently pass. Just out of curiosity, have you never encountered this issue?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -65,9 +65,7 @@ async function generate(html: string, options: CreateOptions): Promise<CreateRes
     await Page.enable(); // Enable Page events
     const url = /^(https?|file|data):/i.test(html) ? html : `data:text/html,${html}`;
     await throwIfCanceled(options);
-    await Page.navigate({url});
-    await throwIfCanceled(options);
-    await Page.loadEventFired();
+    await navigate(client, url);
```
Comment (westy92) [inline src/index.ts]: Could we instead just await on the two in parallel?
```typescript
await Promise.all([Page.navigate({url}), Page.loadEventFired()]); // Resolve order varies
```

Comment (Kriegslustig) [reply src/index.ts]: Oh right, way nicer!

File: src/index.ts [hunk 2]
```diff
@@ -65,9 +65,10 @@ async function generate(html: string, options: CreateOptions): Promise<CreateRes
     await Page.enable(); // Enable Page events
     const url = /^(https?|file|data):/i.test(html) ? html : `data:text/html,${html}`;
     await throwIfCanceled(options);
-    await Page.navigate({url});
-    await throwIfCanceled(options);
-    await Page.loadEventFired();
+    await Promise.all([
+      client.Page.navigate({url}),
```
Comment (westy92) [inline src/index.ts]: Drop the unnecessary `client.`.

File: test/index.ts [hunk 3]
```diff
@@ -196,9 +196,11 @@ describe('HtmlPdf', () => {
             <body>
               <div id="test">Failed!</div>
               <script>
-                setTimeout(() => {
-                  document.getElementById('test').innerHTML = 'Passed!';
-                }, 100);
+                document.addEventListener('DOMContentLoaded', () => {
```
Comment (westy92) [inline test/index.ts]: Why is this change needed?

Comment (Kriegslustig) [reply test/index.ts]: The event listener isn't strictly necessary. But the longer timeout is. When running this test locally on my machine it consistently failed with the new behavior. Only with the increased timeout it passes.

Comment (westy92) [reply test/index.ts]: Please remove the event listener.  Keep the timeout at 100ms and try to increase the test timer to 300ms.

Comment (Kriegslustig) [reply test/index.ts]: I can decrease this timeout to 200ms. If I keep it at 100ms the "permature" test runs after the timeout callback runs. I changed it to 500ms in order to have some tolerance.

Comment (westy92) [reply test/index.ts]: 200ms is okay.  Thanks.

**Comment by westy92** [comment]
Thank you!
